security/advancedtls/crl.go

```diff
@@ -0,0 +1,489 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package advancedtls
```

Comment:
Would it make sense to have a `crl` sub-directory under advancedtls and a new package maybe named `crl`? Currently users will use things like `advancedtls.Config` to initialize CRL-related works, which I think might be a bit misleading because `advancedtls` is not only about CRL checking.
Another alternative might be to have a "CRL" prefix for all public fields, but I guess it's sub-optimal here...

Reply:
I'll rename to RevocationConfig. I can see maybe merging this into a larger advtls.Config similar to golang's tls.Config